src/v/kafka/server/request_context.h

```diff
@@ -248,12 +254,82 @@ class request_context {
         return _conn->server().credentials();
     }
 
+    /**
+     * Submits stored up request authz events to the audit manager
+     *
+     * @param event_type The event type for the message
+     * @param additional_resources In some cases, the authz check validates that
+     * a user has cluster-wide authorization.  In those situations, the
+     * information about the resources is not provided in the authz call.  This
+     * argument is here for those situations where additional resources should
+     * be logged to the audit log so the auditor knows what resources were
+     * modified.
+     *
+     * @return bool - True if enqueued into audit log, false otherwise
+     */
+    template<typename T = model::topic>
```

Comment:
should the default type for T be dropped so call sites are more explicit if they would otherwise have used the default (e.g. audit<model::topic> ...) unless it could be deduced automatically from the additional_resources parameter.